x-pack/plugins/ingest_pipelines/public/application/components/expandable_text.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState, FunctionComponent } from 'react';
+import { EuiLink, EuiText } from '@elastic/eui';
+
+interface Props {
+  text: string;
+  charLimit?: number;
+}
+
+const DEFAULT_CHAR_LIMIT = 25;
+
+export const ExpandableText: FunctionComponent<Props> = ({
+  text,
+  charLimit = DEFAULT_CHAR_LIMIT,
+}) => {
+  const [isExpanded, setExpanded] = useState(false);
+  const exceedsCharLimit = text.length > charLimit;
+
+  const processedText = exceedsCharLimit
+    ? isExpanded
+      ? text
+      : text.substr(0, charLimit) + '...'
+    : text;
+
+  return (
+    <EuiText size="s">
+      {processedText}
+      {exceedsCharLimit && (
+        <>
+          &nbsp;
+          <EuiLink onClick={() => setExpanded(!isExpanded)}>{isExpanded ? 'Less' : 'More'}</EuiLink>
```

Comment:
Generally for updating on previous state I think you are correct. However the [React docs](https://reactjs.org/docs/hooks-state.html#updating-state) give an example of using `setCount(count + 1)` but then we know react batches `setStates` together like [this](https://stackoverflow.com/questions/55510565/do-we-still-need-functional-setstate-way-in-react-hooks). I would say in this particular case I can't see `setExpanded` being called in quick enough succession that it would merit using `prevProps`, but at the same time using `prevProps` is safer, thanks for pointing that out!